src/apply.ml

```diff
@@ -0,0 +1,31 @@
+open Basics
+open Console
+open Terms
+open Refine
+
+(** [nbArgsOf t] counts the number of arguments of the type [t], i.e.
+    it counts the number of Prod binders located at the front of [t]. *)
+let rec nbArgsOf : term -> int = fun t ->
```

Comment:
I think `count_head_products` would be a better name. Note that we use the OCaml naming convention (using `_`) not the Haskell convention (capitalizing first letters of each words).